src/Event/AdminEventExtension.php

```diff
@@ -36,89 +37,155 @@ public function __construct(EventDispatcherInterface $eventDispatcher)
 
     public function configureFormFields(FormMapper $form)
     {
-        $this->eventDispatcher->dispatch(
-            'sonata.admin.event.configure.form',
-            new ConfigureEvent($form->getAdmin(), $form, ConfigureEvent::TYPE_FORM)
-        );
+        if ($this->eventDispatcher instanceof ContractsEventDispatcherInterface) {
+            $event = new ConfigureEvent($form->getAdmin(), $form, ConfigureEvent::TYPE_FORM);
+            $eventName = 'sonata.admin.event.configure.form';
+        } else {
```

Comment:
which will allow us to remove this `else`

Reply:
Oh wait maybe this is irrelevant after you apply [this comment](https://github.com/sonata-project/SonataAdminBundle/pull/5572#issuecomment-498956711)

Reply:
I don't know how to create a boilerplate decorator without using 2 classes based on the existence of the contract, since the signature of the `dispatch()` method is explicitly different in Symfony < 4.3:

https://github.com/symfony/event-dispatcher/blob/a088aafcefb4eef2520a290ed82e4374092a6dff/EventDispatcherInterface.php#L34

Reply:
@OskarStark please advise